modules/apps/object/object-service/src/main/java/com/liferay/object/internal/validation/rule/UniqueCompositeKeyObjectValidationRuleEngineImpl.java

```diff
@@ -139,7 +139,9 @@ private Predicate _getPredicate(
 				Map<String, Object> objectFieldProperties =
 					(Map<String, Object>)entryValues.get(objectField.getName());
 
-				value = String.valueOf(objectFieldProperties.get("key"));
+				if (objectFieldProperties != null) {
```

Comment:
Hey @mateussgn , what happens if objectFieldProperties is null, is there a problem to pass value variable with null value to `ObjectEntrySearchUtil.getUniqueCompositeKeyObjectFieldPredicate((Column<?, Object>)column, objectField.getDBType(), value)`?

Reply:
objectFieldProperties will just exist for picklist, that is why is inside an if statement.
In this case the value of a picklist it is inside of an object nested to objectFieldProperties.
If objectFieldProperties is null, then the value variable still filled with null.
value could be null if you see in line 133, it has been initialized with null.

Reply:
My point is, is there a case where the field is of type picklist and it doesn't have objectFieldProperties? For me it seems to be the case for this NPE bug.
I understand that objectFieldProperties will only exist for picklist, but this was already being checked in an if statement for ObjectFieldConstants.BUSINESS_TYPE_PICKLIST.